src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -89,6 +103,24 @@ class PythonInferSubsystem(Subsystem):
             "treated as a potential dependency if this option is set to 2 but not if set to 3."
         ),
     )
+    assets = BoolOption(
+        "--assets",
+        default=False,
```

Comment:
If you didn't have to consider our deprecation policy, should this default to True? We don't have string imports default to True because of false positives. Not sure what the right behavior is here.

Reply:
Yeah. OK. I can get very behind making `True` the default.

Reply:
Ah, but note that we do need to consider our deprecation policy, I think. Possibly. Unclear to me. I was only asking what the aspiration is.

In the interest of landing this for 2.10, how about using `False` for now, having a conversation in Slack or dedicated PR on Monday, and then possibly changing it in time for 2.10? That'd be good anyways to have two entries in changelog:

```
# new feature
* Add `[python-infer].assets` to infer resources and files from strings

# user-api changes
* Enable `[python-infer].assets` by default
```